app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/masterpasswordgenerator/MasterPasswordGeneratorViewModelTest.kt

```diff
@@ -0,0 +1,150 @@
+
+            verify { mockPolicyManager.getActivePolicies(type = PolicyTypeJson.MASTER_PASSWORD) }
+        }
+
+    @Test
+    fun `State updates when generate action is sent and repository returns success result`() =
+        runTest {
+            val viewModel = createViewModel()
+            val newPassPhrase = "I am new"
+            val expectedResult = MasterPasswordGeneratorState(generatedPassword = newPassPhrase)
+
+            viewModel.stateFlow.test {
+                assertNotEquals(expectedResult, awaitItem())
+                fakeGeneratorRepository.setMockGeneratePassphraseResult(
+                    GeneratedPassphraseResult.Success(generatedString = newPassPhrase),
+                )
+                viewModel.trySendAction(
+                    MasterPasswordGeneratorAction.GeneratePasswordClickAction,
+                )
+                assertEquals(expectedResult, awaitItem())
+            }
+        }
+
+    @Test
+    fun `ShowSnackbar event is sent when the generate passphrase result is a failure`() = runTest {
+        val viewModel = createViewModel()
+
+        fakeGeneratorRepository.setMockGeneratePassphraseResult(
+            GeneratedPassphraseResult.InvalidRequest,
+        )
+
+        viewModel.eventFlow.test {
+            viewModel.trySendAction(MasterPasswordGeneratorAction.GeneratePasswordClickAction)
+            assertEquals(
+                MasterPasswordGeneratorEvent.ShowSnackbar(R.string.an_error_has_occurred.asText()),
+                awaitItem(),
+            )
+        }
+    }
+
+    @Test
+    fun `NavigateBack event is sent when BackClickAction is handled`() = runTest {
+        val viewModel = createViewModel()
+
+        viewModel.eventFlow.test {
+            viewModel.trySendAction(MasterPasswordGeneratorAction.BackClickAction)
+            assertEquals(MasterPasswordGeneratorEvent.NavigateBack, awaitItem())
+        }
+    }
+
+    @Test
+    fun `NavigateToPreventLockout event is sent when PreventLockoutClickAction is handled`() =
+        runTest {
+            val viewModel = createViewModel()
+
+            viewModel.eventFlow.test {
+                viewModel.trySendAction(MasterPasswordGeneratorAction.PreventLockoutClickAction)
+                assertEquals(MasterPasswordGeneratorEvent.NavigateToPreventLockout, awaitItem())
+            }
+        }
+
+    // region: helpers
+
+    private fun createViewModel(
+        savedStateHandle: SavedStateHandle = createSavedStateHandle(),
+    ): MasterPasswordGeneratorViewModel = MasterPasswordGeneratorViewModel(
+        savedStateHandle = savedStateHandle,
+        generatorRepository = fakeGeneratorRepository,
+        policyManager = mockPolicyManager,
+    )
+
+    private fun createSavedStateHandle(initialState: MasterPasswordGeneratorState? = null) =
+        SavedStateHandle().apply {
+            initialState?.let {
+                this["state"] = it
+            }
+        }
+
+// end region: helpers
```

Comment:
This does not work as a region unless it is one word: `endregion`.
Also, can you indent it and remove the colons